src/ec/packages/ec-component-site-header-harmonised/ecl-site-header-harmonised.html.twig

```diff
@@ -136,12 +120,57 @@
   {% set _search_toggle = _search_toggle|merge(search_toggle) %}
 {% endif %}
 
-{% if search_form.text_input is defined and search_form.text_input is not empty %}
-  {% set _search_form = _search_form|merge({ text_input: _search_form.text_input|merge(search_form.text_input) }) %}
-{% endif %}
+{% if search_form is defined and search_form is not empty %}
+  {% if search_form.extra_attributes is defined and search_form.extra_attributes is not empty %}
+    {% set _search_form_extra_attributes = _search_form_extra_attributes|merge(search_form.extra_attributes) %}
+  {% endif %}
+  {% if search_form.extra_classes is defined and search_form.extra_classes is not empty %}
+    {% set _search_form_extra_classes = _search_form_extra_classes ~ ' ' ~ search_form.extra_classes %}
+  {% endif %}
+
+  {% if search_form.text_input is defined and search_form.text_input is not empty %}
+    {% if search_form.text_input.extra_classes is defined and search_form.text_input.extra_classes is not empty %}
+      {% set _search_input_extra_classes = _search_input_extra_classes ~ ' ' ~ _search_form.text_input.extra_classes %}
+    {% endif %}
+    {% if search_form.text_input.extra_attributes is defined and search_form.text_input.extra_attributes is not empty %}
+      {% set _search_input_extra_attributes = _search_input_extra_attributes|merge(search_form.text_input.extra_attributes) %}
+    {% endif %}
+  {% endif %}
+
+  {% if search_form.button is defined and search_form.button is not empty %}
+    {% if search_form.button.extra_classes is defined and search_form.button.extra_classes is not empty %}
+      {% set _search_button_extra_classes = _search_button_extra_classes ~ ' ' ~ search_form.button.extra_classes %}
+    {% endif %}
+    {% if search_form.button.extra_attributes is defined and search_form.button.extra_attributes is not empty %}
+      {% set _search_button_extra_attributes = _search_button_extra_attributes|merge(search_form.button.extra_attributes) %}
+    {% endif %}
+    {% if search_form.button.label is defined and search_form.button.label is not empty %}
+      {% set _search_button_label = search_form.button.label %}
+    {% endif %}
+  {% endif %}
 
-{% if search_form.button is defined and search_form.button is not empty %}
-  {% set _search_form = _search_form|merge({ button: _search_form.button|merge(search_form.button) }) %}
+  {% set _search_form = {
+    text_input: search_form.text_input|default({})|merge({
+      extra_classes: _search_input_extra_classes,
+      extra_attributes: _search_input_extra_attributes,
+    }),
+    button: {
+      label: _search_button_label,
+      variant: 'search',
+      icon: {
+        type: 'general',
+        name: 'search',
```

Comment:
I think this can be removed like you did in Standardized?

Reply:
yep